src/vsr/replica.zig

```diff
@@ -4081,6 +4081,28 @@ pub fn ReplicaType(
             }
         }
 
+        fn commit_register(
+            self: *Self,
+            prepare: *const Message.Prepare,
+            output_buffer: *align(16) [constants.message_body_size_max]u8,
+        ) usize {
+            assert(self.commit_stage == .setup_client_replies);
+            assert(self.commit_prepare.? == prepare);
+            assert(prepare.header.command == .prepare);
+            assert(prepare.header.operation == .register);
+            assert(prepare.header.size == @sizeOf(vsr.Header));
+            assert(prepare.header.op == self.commit_min + 1);
+            assert(prepare.header.op <= self.op);
+
+            const result = std.mem.bytesAsValue(
+                vsr.RegisterResult,
+                output_buffer[0..@sizeOf(vsr.RegisterResult)],
+            );
+
+            result.* = .{};
```

Comment:
Client should assert that the result is zeroed when receiving a register reply

Reply:
:thinking: I actually left that out deliberately.. My thinking was that at some point we will change the register result to be non-zero, and we will want to do that without crashing old clients.